product-service/handler.ts

```diff
@@ -0,0 +1,9 @@
+import { ProductService } from './src/services/productService';
+
+import { getProductByIdHandler } from './src/getProductById';
+import { getAllProductsHandler } from './src/getAllProducts';
+
+const productService = new ProductService();
```

Comment:
[minor]

I assume that the reason that you move this service initialization this way and pass it as the parameter - you are trying to organize something like dependency inversion. Basically it makes sense from testing point of view, of course.
But it's quite a high level of abstraction and since JS is dynamically typed and can be easily mocked - it's not so essential to provide such mechanisms.
To recap - it's completely correct to use it, but you can also require and initialize services in direct handlers and it's also be valid.